hazelcast/src/main/java/com/hazelcast/client/impl/ClientEngineImpl.java

```diff
@@ -588,6 +590,32 @@ public Operation getPostJoinOperation() {
         return resultMap;
     }
 
+    @Override
+    public List<Map.Entry<String, List<Map.Entry<String, String>>>> getClientStatistics() {
+        Collection<ClientEndpoint> clientEndpoints = endpointManager.getEndpoints();
+        List<Map.Entry<String, List<Map.Entry<String, String>>>> statsList = new ArrayList<Map.Entry<String, List<Map.Entry<String, String>>>>(
+                clientEndpoints.size());
+        for (final ClientEndpoint e : clientEndpoints) {
+            statsList.add(new Map.Entry<String, List<Map.Entry<String, String>>>() {
```

Comment:
Instead of Entry interface use AbstractMap.SimpleEntry it also has correct hashcode and equals implementations which is missing here.